feconf.py

```diff
@@ -61,8 +61,23 @@
     FRONTEND_TEMPLATES_DIR = os.path.join('core', 'templates', 'dev', 'head')
 DEPENDENCIES_TEMPLATES_DIR = (
     os.path.join(EXTENSIONS_DIR_PREFIX, 'extensions', 'dependencies'))
+
+if not constants.DEV_MODE:
+    VALUE_GENERATORS_DIR_FOR_JS = os.path.join(
+        'local_compiled_js', 'extensions', 'value_generators')
+else:
+    VALUE_GENERATORS_DIR_FOR_JS = os.path.join(
+        'extensions', 'value_generators')
 VALUE_GENERATORS_DIR = os.path.join('extensions', 'value_generators')
+
+if not constants.DEV_MODE:
```

Comment:
Ditto.

Reply:
For visualizations dir, it is fine to have just this code:

```
if not constants.DEV_MODE:
    VISUALIZATIONS_DIR = os.path.join(
        'local_compiled_js', 'extensions', 'visualizations')
else:
    VISUALIZATIONS_DIR = os.path.join(
        'extensions', 'visualizations')
```

This constant is only used [here](https://github.com/oppia/oppia/blob/develop/core/domain/visualization_registry.py#L55) to obtain the js file.

Reply:
See above.

Reply:
Same as previous